viewer/packages/tools/src/Measurement/MeasurementTool.ts

```diff
@@ -0,0 +1,445 @@
+  }
+
+  /**
+   * Exit measurement mode.
+   */
+  exitMeasurementMode(): void {
+    if (!this._measurementMode) {
+      throw new Error('Measurement mode is not active, call enterMeasurementMode()');
+    }
+    if (this._activeMeasurement) {
+      this.cancelActiveMeasurement();
+    }
+    this._viewer.off('click', this._handlePointerClick);
+    this._events.measurementEnded.fire();
+    this._measurementMode = false;
+  }
+
+  /**
+   * Removes a measurement from the Cognite3DViewer.
+   * @param measurement Measurement to be removed from Cognite3DViewer.
+   */
+  removeMeasurement(measurement: Measurement): void {
+    const index = this._measurements.findIndex(
+      measurementManager => measurementManager.getMeasurement() === measurement
+    );
+    if (index > -1) {
+      this._measurements[index].removeMeasurement();
+      this._measurements.splice(index, 1);
+      this._viewer.requestRedraw();
+    } else {
+      throw new Error('Measurement not found');
+    }
+  }
+
+  /**
+   * Removes all measurements from the Cognite3DViewer.
+   */
+  removeAllMeasurements(): void {
+    //clear all mesh, geometry & labels.
+    this._measurements.forEach(measurement => {
+      measurement.removeMeasurement();
+    });
+    this._measurements.splice(0);
+    this._viewer.requestRedraw();
+  }
+
+  /**
+   * Sets the visiblity of labels in the Measurement.
+   * @param enable
+   */
+  setMeasurementLabelsVisible(enable: boolean): void {
+    this._htmlOverlay.visible(enable);
+  }
+
+  /**
+   * Sets Measurement line width, color and label units value for the next measurment.
+   * @param options MeasurementOptions to set line width, color and callback for custom operation on measurement labels.
+   */
+  setLineOptions(options: MeasurementOptions): void {
+    this._options = {
+      ...this._options,
+      ...options
+    };
+  }
+
+  /**
+   * Update selected line width.
+   * @param measurement Measurement to be updated.
+   * @param lineWidth Width of the measuring line.
+   */
+  updateLineWidth(measurement: Measurement, lineWidth: number): void {
+    const index = this._measurements.findIndex(
+      measurementManager => measurementManager.getMeasurement() === measurement
+    );
+    if (index > -1) {
+      this._measurements[index].updateLineWidth(lineWidth);
+      this._viewer.requestRedraw();
+    } else {
+      throw new Error('Measurement not found');
+    }
+  }
+
+  /**
+   * Update selected line color.
+   * @param measurement Measurement to be updated.
+   * @param color Color of the measuring line.
+   */
+  updateLineColor(measurement: Measurement, color: THREE.Color): void {
+    const index = this._measurements.findIndex(
+      measurementManager => measurementManager.getMeasurement() === measurement
+    );
+    if (index > -1) {
+      this._measurements[index].updateLineColor(color);
+      this._viewer.requestRedraw();
+    } else {
+      throw new Error('Measurement not found');
+    }
```

Comment:
```suggestion
    if (index === -1) {
      throw new Error('Measurement not found');
    }
    this._measurements[index].updateLineColor(color);
    this._viewer.requestRedraw();
```

Reply:
Have applied the suggestion for all instances